### PR TITLE
Add functional test micro-framework

### LINKED ISSUE #26: Add functional tests
With more and more moving bits, the project has reached a state in which we can no longer rely on the unit tests only. We should set up and gradually add functional tests, that will help us gain more confidence in changing existing features as well as adding new ones.

### PR DESCRIPTION
**Pull request by thilp**
This is a micro-framework for running arbitrary scripts in a clean environment where the `transformer` package is installed.

~Each test case is a script (Python, bash, and sh supported now, but it's trivial to modify this list) which name starts with `test-` and that is supposed to exit with 0 if (and only if) the corresponding feature works.~
Each test case is a Pytest script.

~A special script, `run.py`,~ `make` is the entrypoint of the test suite:
- it creates a fresh virtualenv,
- it builds and install Transformer (with `poetry build`) in that virtualenv,
- it runs all test scripts in the context of that environment, keeping track of the number of failures,
- it exits with 0 if and only if at least one test was run and all tests run were successful,
- ~it copiously logs everything happening along the way.~

This PR comes with a first test that ensures the installed `transformer` script can display its own version.

Close #26.

## Types of Changes

_What types of changes does your code introduce? Keep the ones that apply:_

- New feature (non-breaking change which adds functionality)
- Documentation / non-code

## Tasks

_List of tasks you will do to complete the PR:_

- [x] Run these functional tests as part as the CI pipeline.
- [x] Make it easy to run these tests without knowing anything about the framework: include in Makefile
- [x] Update CHANGELOG.md

## Review

_List of tasks the reviewer must do to review the PR:_

- Try the framework.
- Should we add more tests in this PR? Do you have ideas about which ones?

### CODE REVIEW

**Comment by thilp** [comment]
The functional test suite running in Travis can be seen [**here**](https://travis-ci.org/zalando-incubator/Transformer/jobs/499326706#L343-L359).

**Comment by tortila** [comment]
As discussed offline, I'm in favour of using existing tools and standards for orchestrating the dependencies as well as running the tests. I see why we could get some benefits of a custom solution like this, however I think that we don't need it just yet ([YAGNI](https://en.wikipedia.org/wiki/You_aren%27t_gonna_need_it)) and it would be simpler to start with a non-custom solution.

**Comment by thilp** [comment]
The **new** functional test suite running in Travis can be seen [**here**](https://travis-ci.org/zalando-incubator/Transformer/jobs/499360292#L348-L401).

File: functional-tests/README.md [hunk 1]
```diff
@@ -0,0 +1,18 @@
+Functional tests for Transformer's CLI.
+
+They run in a special virtualenv where the `transformer` package, created
+by `poetry build`, is installed.
+
+This directory contains:
+  - a Makefile,
+  - Python scripts with a name starting with `test_`.
```
Comment (tortila) [inline functional-tests/README.md]: I think that we can just state what test runner is used, the rest of it (test scripts starting with `test_` - here and in the paragraphs below) to me is only a detail of Pytest (its default configuration).

File: functional-tests/README.md [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+The Makefile sets up the appropriate environment: it builds transformer,
+creates a virtualenv, installs transformer in that virtualenv, and finally runs
+the `test_*` scripts with Pytest in that environment.
+
+`test_*` scripts can be stored in arbitrary subdirectories, as long as these
+also have a name starting with `test_`.
+
+`make` will exit with 0 if (and only if) at least one test was run and all
+of them were successful.
```
Comment (tortila) [inline functional-tests/README.md]: Am I right assuming that this is also an implementation detail of Pytest? If so, I would suggest not documenting it at all.

**Comment by thilp** [comment]
:+1:

**Pull request merged**